catalog/task/create-pyxis-image/0.1/create-pyxis-image.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-pyxis-image
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that executes a python script to push a container image's metadata to Pyxis
+  params:
+    - name: pyxisUrl
+      type: string
+      description: The URL to the Pyxis API to be used
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: certified
+      type: string
+      description: If set to true, the image will be marked as certified in its Pyxis entry
+      default: "false"
+    - name: tag
+      type: string
+      description: The tag to use when pushing the container image metadata to Pyxis
+    - name: isLatest
+      type: string
+      description: If set to true, the image will have a latest tag added with its Pyxis entry
+      default: "false"
+    - name: skopeoInspectFile
+      type: string
+      description: The filename of the saved skopeo inspect output stored on the input workspace
+      default: "skopeo-inspect.json"
+  results:
+    - name: containerImageID
+      description: ID of the created entry in Pyxis
+  workspaces:
+    - name: input
+      description: The workspace where the skopeo-inspect.json file is located
+  steps:
+    - name: create-pyxis-image
+      image:
+        quay.io/hacbs-release/release-utils@sha256:1cb213e052adc0919f0ef1eb7f0fbb0f1a3b87cb98024095d1e1d036e8f5c5ac
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
+      script: |
+        if [[ -z "$(params.tag)" ]]
+        then
+          echo "Required parameter `tag` not provided"
+          exit 1
+        fi
+
+        echo "${pyxisCert}" > /tmp/crt
+        echo "${pyxisKey}" > /tmp/key
+
+        set -x
+
+        pushd /home/release-utils/pyxis
+        PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key python3 create_container_image.py \
+          --pyxis-url $(params.pyxisUrl) \
+          --certified $(params.certified) \
+          --tag $(params.tag) \
+          --is-latest $(params.isLatest) \
+          --verbose \
+          --skopeo-result $(workspaces.input.path)/$(params.skopeoInspectFile) | tee /tmp/output
+
+        cat /tmp/output | grep 'The image id is' | awk '{print $NF}' | tee $(results.containerImageID.path)
```

Comment:
The line is already printed as part of the python script execution above, so no need to print the plain id again here, right? Meaning that we don't need the `tee` here and can write directly to the results file.

But also, it seems that this line is only printed if the image already existed in pyxis: https://github.com/hacbs-release/release-utils/blob/main/pyxis/create_container_image.py#L78

But creating the image for the first time is also a valid use case and we need to get it somehow in such case as well, no? Or am I missing something?

Reply:
I slightly cleaned this up, but the script prints out more than just the id, and I want *only* that as the result, that is why I have this line.

Good catch on it only being printed if it already exists, I meant to change that in the python script but must not have committed it. I'll open a PR for that

Reply:
> I slightly cleaned this up, but the script prints out more than just the id, and I want only that as the result, that is why I have this line.

What I meant is that we don't need the plain id in the log output (after it has been already printed out to the log as part of the python script).

So you'll have something like this in the log:
```
..
The image id is 123456
..
123456
```
What I'm saying is that we don't have to repeat the id in the log again, so the last line could be something like:
```sh
cat /tmp/output | grep 'The image id is' | awk '{print $NF}' > $(results.containerImageID.path)
```
But it's not a big deal, it can also stay as is.

> Good catch on it only being printed if it already exists, I meant to change that in the python script but must not have committed it. I'll open a PR for that

OK, great. Can you link the PR from here and then we can resolve this thread so that it's not forgotten?

Reply:
https://github.com/hacbs-release/release-utils/pull/12 is the PR. It will be changed from ` | tee` to `>` in the next push

Reply:
fixed